### PR TITLE
ORC-801: Clean up Logging

### PR DESCRIPTION
**Pull request by belugabehr**
### What changes were proposed in this pull request?
Clean up some logging statements with the project. Remove instances of log-and-throw anti-pattern, remove superfluous logging guards, use anchors in debug statements instead of concatenation.

### Why are the changes needed?
Trivial performance increases, decrease in lines of code (readability), log-and-throw makes troubleshooting more difficult because the same error may be logged multiple times.

### How was this patch tested?
Existing unit tests, but should be no change in functionality.

### CODE REVIEW

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Hi, @belugabehr . Thank you for making a PR.
Could you fill the PR template?

**Comment by belugabehr** [comment]
@dongjoon-hyun PR template completed.  Thanks!

File: java/core/src/java/org/apache/orc/impl/RecordReaderImpl.java [hunk 1]
```diff
@@ -1030,13 +1026,11 @@ public SargApplier(SearchArgument sarg,
...
-                        " Skipping ORC PPD." +
-                        " Stats: " + stats +
-                        " Predicate: " + predicate;
-                    LOG.warn(reason, e);
-                  }
+                  final String reason = e.getClass().getSimpleName() + " when evaluating predicate." +
+                      " Skipping ORC PPD." +
+                      " Stats: " + stats +
+                      " Predicate: " + predicate;
+                  LOG.warn(reason, e);
```
Comment (dongjoon-hyun) [inline java/core/src/java/org/apache/orc/impl/RecordReaderImpl.java:1033]: Shall we revert this change?

Comment (belugabehr) [reply java/core/src/java/org/apache/orc/impl/RecordReaderImpl.java:1033]: Hello. This change is intentional.  There is never a reason to wrap a `warn` level message in logging guards: `isWarnEnabled`.  The answer will always be "true"

Comment (dongjoon-hyun) [reply java/core/src/java/org/apache/orc/impl/RecordReaderImpl.java:1033]: Ah, got it. It makes sense.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @belugabehr .
Merged to master for Apache ORC 1.7.0.
